From d45073cf8eeb706b29e35c0a72df30d7c19ea4c1 Mon Sep 17 00:00:00 2001
From: Felix Pehla <29adc1fd92@gmail.com>
Date: Thu, 27 Nov 2025 10:15:51 +0100
Subject: [mkinitcpio] [PATCH 1/2] install/systemd: add libseccomp as
 dependency

Since systemd v259, libseccomp will now be opened via dlopen(), so it is
no longer automatically detected as a dependency. This is still used by
systemd-udevd.service, and will cause a warning to be printed in the
logs if not present.
---
 install/systemd | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/install/systemd b/install/systemd
index 912fedf..5e91617 100644
--- a/install/systemd
+++ b/install/systemd
@@ -142,7 +142,8 @@ build() {
     #  tss2-*   -> tpm2-util
     #  kmod     -> systemd-shared
     #  qrencode -> bsod
-    for LIB in kmod tss2-{esys,rc,mu,tcti-'*'} qrencode; do
+    #  seccomp  -> systemd
+    for LIB in kmod tss2-{esys,rc,mu,tcti-'*'} qrencode seccomp; do
         LC_ALL=C.UTF-8 find /usr/lib/ -maxdepth 1 -name "lib${LIB}.so*" | while read -r FILE; do
             if [[ -L "${FILE}" ]]; then
                 add_symlink "${FILE}"

From cf31d75ec8cb420dec951d2cc102088a181b213a Mon Sep 17 00:00:00 2001
From: Felix Pehla <29adc1fd92@gmail.com>
Date: Thu, 27 Nov 2025 10:22:26 +0100
Subject: [mkinitcpio] [PATCH 2/2] install/systemd: add libacl as dependency

Since systemd v259, libacl will now be opened via dlopen(), so it is no
longer automatically detected as a dependency. This is still used by
systemd-tmpfiles if ACLs are specified in one of its configs, which is
the case for the default systemd tmpfiles. Missing libacl.so will cause
warnings to be written to the log.
---
 install/systemd | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/install/systemd b/install/systemd
index 5e91617..fed9eb9 100644
--- a/install/systemd
+++ b/install/systemd
@@ -143,7 +143,8 @@ build() {
     #  kmod     -> systemd-shared
     #  qrencode -> bsod
     #  seccomp  -> systemd
-    for LIB in kmod tss2-{esys,rc,mu,tcti-'*'} qrencode seccomp; do
+    #  acl      -> systemd-tmpfiles
+    for LIB in kmod tss2-{esys,rc,mu,tcti-'*'} qrencode seccomp acl; do
         LC_ALL=C.UTF-8 find /usr/lib/ -maxdepth 1 -name "lib${LIB}.so*" | while read -r FILE; do
             if [[ -L "${FILE}" ]]; then
                 add_symlink "${FILE}"
